seqr/views/apis/saved_variant_api_tests.py

```diff
@@ -731,3 +732,127 @@ def test_update_variant_main_transcript(self):
         self.assertEqual(saved_variant.selected_main_transcript_id, transcript_id)
 
 
+# Tests for AnVIL access disabled
+class LocalSavedVariantAPITest(AuthenticationTestCase, TestHelpers.SavedVariantAPITest):
+    fixtures = ['users', '1kg_project']
+
+
+def assert_no_list_ws_has_acl(self, acl_call_count):
+    self.mock_list_workspaces.assert_not_called()
+    self.mock_service_account.get.assert_called_with(
+        'api/workspaces/my-seqr-billing/anvil-1kg project n\u00e5me with uni\u00e7\u00f8de/acl')
+    self.assertEqual(self.mock_service_account.get.call_count, acl_call_count)
+
+
+# Test for permissions from AnVIL only
+class AnvilSavedVariantAPITest(AnvilAuthenticationTestCase, TestHelpers.SavedVariantAPITest):
+    fixtures = ['users', 'social_auth', '1kg_project']
+
+    def test_saved_variant_data(self):
+        super(AnvilSavedVariantAPITest, self).test_saved_variant_data()
+        assert_no_list_ws_has_acl(self, 5)
+
+    def test_create_saved_variant(self):
+        super(AnvilSavedVariantAPITest, self).test_create_saved_variant()
+        assert_no_list_ws_has_acl(self, 2)
+
+    def test_create_saved_sv_variant(self):
+        super(AnvilSavedVariantAPITest, self).test_create_saved_sv_variant()
+        assert_no_list_ws_has_acl(self, 2)
+
+    def test_create_saved_compound_hets(self):
+        super(AnvilSavedVariantAPITest, self).test_create_saved_compound_hets()
+        assert_no_list_ws_has_acl(self, 2)
+
+    def test_create_update_and_delete_variant_note(self):
+        super(AnvilSavedVariantAPITest, self).test_create_update_and_delete_variant_note()
+        assert_no_list_ws_has_acl(self, 6)
+
+    def test_create_partially_saved_compound_het_variant_note(self):
+        super(AnvilSavedVariantAPITest, self).test_create_partially_saved_compound_het_variant_note()
+        assert_no_list_ws_has_acl(self, 2)
+
+    def test_create_update_and_delete_compound_hets_variant_note(self):
+        super(AnvilSavedVariantAPITest, self).test_create_update_and_delete_compound_hets_variant_note()
+        assert_no_list_ws_has_acl(self, 7)
```

Comment:
why doe the pure AnVIL tests call the service account api many times, but the mix test case always calls it exactly once?

Reply:
The pure AnVIL tests fetch the permissions from AnVIL every time while the mix cases don't have to access AnVIL because they get the permissions local already.